### PR TITLE
add descriptions for background pictures

### LINKED ISSUE #40689: [BITV]: Replace `aria-label`'s of background buttons with another name
This is accessibility issue.

For now background buttons have `aria-label`'s which are not really understandable, like "Soft floral (Hannah MacLean, CC0)". The `aria-label` have to be like "Background with abstract ..."

Each name have to unique.

Screenshot from 2023-09-28 18-51-42

### PR DESCRIPTION
**Pull request by szaimen**
Close https://github.com/nextcloud/server/issues/40689

I added translations for the attributions and added descriptions for each foto.

### CODE REVIEW

**Comment by szaimen** [comment]
```
<html>
<body>
<!--StartFragment-->

There was 1 error:
--
218 |  
219 | 1) OCA\Theming\Tests\ThemingDefaultsTest::testGetColorPrimaryWithCustomBackground
220 | Undefined array key 2
221 |  
222 | /drone/src/apps/theming/tests/ThemingDefaultsTest.php:478

<!--EndFragment-->
</body>
</html>
```

File: apps/theming/lib/Service/BackgroundService.php [hunk 1]
```diff
@@ -257,4 +158,129 @@ private function getAppDataFolder(): ISimpleFolder {
 			return $rootFolder->newFolder($this->userId);
 		}
 	}
+
+	public function getShippedBackgrounds(): array {
+		return [
+			'hannah-maclean-soft-floral.jpg' => [
+				'attribution' => $this->l10n->t('Soft floral (Hannah MacLean, CC0)'),
+				'description' => $this->l10n->t('Abstract picture in yellow and white color whith a flower on it'),
```
Comment (szaimen) [inline apps/theming/lib/Service/BackgroundService.php:166]: For reviewers: Please review each description carefully. Does the text look fine?

File: apps/theming/lib/Themes/CommonThemeTrait.php [hunk 2]
```diff
@@ -31,6 +31,13 @@
 
 trait CommonThemeTrait {
 	public Util $util;
+	private BackgroundService $backgroundService;
+
+	public function __construct(
+		BackgroundService $backgroundService,
+	) {
+		$this->backgroundService = $backgroundService;
+	}
```
Comment (szaimen) [inline apps/theming/lib/Themes/CommonThemeTrait.php:40]: apparently this is not working currently as it is a trait and not a class...

Comment (szaimen) [reply apps/theming/lib/Themes/CommonThemeTrait.php:40]: I got `"Typed property OCA\\\\\\\\Theming\\\\\\\\Themes\\\\\\\\DefaultTheme::$backgroundService must not be accessed before initialization in file '/var/www/nextcloud/apps/theming/lib/Themes/CommonThemeTrait.php`

Comment (szaimen) [reply apps/theming/lib/Themes/CommonThemeTrait.php:40]: So apparently the constructor does not work here. Question then how to make it work at all?

Comment (ChristophWurst) [reply apps/theming/lib/Themes/CommonThemeTrait.php:40]: Pass the dependency as an argument of the methods that need it. Example: https://github.com/nextcloud/server/blob/master/lib/public/AppFramework/Db/TTransactional.php.

The concept of DI and properties for dependencies doesn't really fit traits.

File: apps/theming/lib/Themes/DefaultTheme.php [hunk 3]
```diff
@@ -57,7 +58,8 @@ public function __construct(Util $util,
 								ImageManager $imageManager,
 								IConfig $config,
 								IL10N $l,
-								IAppManager $appManager) {
+								IAppManager $appManager,
+								BackgroundService $backgroundService) {
```
Comment (szaimen) [inline apps/theming/lib/Themes/DefaultTheme.php:62]: So this is now breaking many tests (34 afaics). @ChristophWurst Any other idea how I can do this without having to refactor everything? Adding backgroundservice to this constructor at least does not seem to work...

Comment (ChristophWurst) [reply apps/theming/lib/Themes/DefaultTheme.php:62]: I don't have any other ideas.

Can you adjust/fix the tests?

Comment (szaimen) [reply apps/theming/lib/Themes/DefaultTheme.php:62]: What if I would move the commonthemetrait? Do you think this would make the logic easier?

Comment (ChristophWurst) [reply apps/theming/lib/Themes/DefaultTheme.php:62]: Try

Comment (szaimen) [reply apps/theming/lib/Themes/DefaultTheme.php:62]: Lets see https://github.com/nextcloud/server/pull/41235

Comment (ChristophWurst) [reply apps/theming/lib/Themes/DefaultTheme.php:62]: Could I ask you to try your changes locally? Using CI for trial and error is really expensive and slows down real builds.

**Comment by szaimen** [comment]
Superceded by https://github.com/nextcloud/server/pull/41256

**Pull request closed without merging**